taker-frontend/src/App.tsx

```diff
@@ -69,40 +67,26 @@ export const App = () => {
     const [source, isConnected] = useEventSource("/api/feed");
     const walletInfo = useLatestEvent<WalletInfo>(source, "wallet");
 
-    const makerLong = useLatestEvent<Order>(source, "long_offer", intoOrder);
-    const makerShort = useLatestEvent<Order>(source, "short_offer", intoOrder);
-
-    const shortOrder = {
-        id: makerLong?.id,
-        initialFundingFeePerParcel: makerLong?.initial_funding_fee_per_parcel,
-        marginPerParcel: makerLong?.margin_per_parcel,
-        price: parseOptionalNumber(makerLong?.price),
-    };
-
-    const longOrder = {
-        id: makerShort?.id,
-        initialFundingFeePerParcel: makerShort?.initial_funding_fee_per_parcel,
-        marginPerParcel: makerShort?.margin_per_parcel,
-        price: parseOptionalNumber(makerShort?.price),
-    };
+    const makerLong = useLatestEvent<MakerOffer>(source, "long_offer", intoMakerOffer);
+    const makerShort = useLatestEvent<MakerOffer>(source, "short_offer", intoMakerOffer);
 
-    const globalTradeParams = extractGlobalTradeParams(makerLong, makerShort);
+    const shortOffer = makerOfferToTakerOffer(makerLong);
+    const longOffer = makerOfferToTakerOffer(makerShort);
 
-    function extractGlobalTradeParams(long: Order | null, short: Order | null) {
-        const order = long ? long : short ? short : null;
-
-        if (order) {
+    function makerOfferToTakerOffer(offer: MakerOffer | null): Offer {
+        if (offer) {
             return {
-                minQuantity: parseOptionalNumber(order.min_quantity) || 0,
-                maxQuantity: parseOptionalNumber(order.max_quantity) || 0,
-                parcelSize: parseOptionalNumber(order.parcel_size) || 0,
-                leverage: order.leverage || 0,
-                openingFee: order.opening_fee || 0,
-                fundingRateAnnualized: order.funding_rate_annualized_percent || "0",
-                fundingRateHourly: order
-                    ? Number.parseFloat(order.funding_rate_hourly_percent).toFixed(5)
-                    : "0",
-                liquidationPrice: parseOptionalNumber(order.liquidation_price),
+                id: offer.id,
+                initialFundingFeePerParcel: offer.initial_funding_fee_per_parcel,
+                marginPerParcel: offer.margin_per_parcel,
+                price: Number.parseFloat(offer.price),
```

Comment:
not particular related to your PR: 

why are `offer.price`, `offer.min_/max_quantity` strings in the first place? Couldn't we just say these are numbers and avoid `Number.parseFloat`?

Reply:
I think we could do that - we are probably a bit too "correct" there. In the daemon they are represented as `Decimal`, and we send them out as string. We *could* use a `Decimal` library in the UI code as well, but don't at the moment, so for now we could just just make them numbers I support :)

Reply:
I gave this a try, please have another look

Reply:
I think it was a historical decision to control some of the precision-handling in the projection (which exists solely for the purpose of the UI)...